### ISSUE TITLE
sphinx doctest

### ISSUE THREAD

**Issue by davidsean** [espressomd/espresso#1849]
Test the code-blocks / snippets for (at the very least) correct syntax.

**Comment by KaiSzuttor** [comment]
http://www.sphinx-doc.org/en/stable/ext/doctest.html

**Comment by davidsean** [comment]
Yes (hence why I called this issue doctest). Someone needs to implement it, with espresso-related preambles and cleanup current snippets that have imports.

**Comment by RudolfWeeber** [comment]
> Yes (hence why I called this issue doctest). Someone needs to implement it, with espresso-related preambles and cleanup current snippets that have imports.
Not sure if I understand this correctly. I think, we should restrict
ourselvs to syntactic checking. E.g., becuase there are placeholders
for user specific values in some snippets. 
In some places, getting rid of tthem would make the docs more
difficult.
Syntactic correctness can still be achieved and checked by using valid variable names
for the place holders.

**Comment by KaiSzuttor** [comment]
I think it makes sense to check if the samples work if they are copy pasted to an empty textfile.

**Comment by RudolfWeeber** [comment]
On Wed, Feb 14, 2018 at 02:12:11PM +0000, Kai Szuttor wrote:
> I think it makes sense to check if the samples work if they are copy pasted to an empty textfile.
Many samples are not formulated in this way. E.g.,
http://espressomd.org/html/doc/advanced_methods.html#creating-bonds-when-particles-collide
but I'm sure there are a lots of others.
I'm not sure if inserting numbers into those samples would improve
things. But syntatic correcntess is surely worth checking.

**Comment by KaiSzuttor** [comment]
i was not here at the meeting yesterday, @KonradBreitsprecher  mentioned that this was discussed. what was the outcome?

**Comment by KaiSzuttor** [comment]
currently, we have a target for doctest but we actually do not run the check in CI (because it's broken probably).
